pkg/scripts_test/install_test.go

```diff
@@ -99,21 +134,65 @@ func exitCode(cmd *exec.Cmd) (int, error) {
 	return 0, fmt.Errorf("cannot obtain exit code: %v", err)
 }
 
-func TestInstallation(t *testing.T) {
-	defer tearDown(t)
-
-	_, err := os.Stat("/usr/local/bin/otelcol-sumo")
-	require.ErrorIs(t, err, os.ErrNotExist, "/usr/local/bin/otelcol-sumo is already created")
-	runScript(t)
+func checkBinaryCreated(t *testing.T, opt installOptions) {
+	_, err := os.Stat(binaryPath)
+	require.NoError(t, err, "binary has not been created")
+}
 
-	_, err = os.Stat("/usr/local/bin/otelcol-sumo")
-	require.NoError(t, err, "/usr/local/bin/otelcol-sumo has not been created")
+func checkBinaryNotCreated(t *testing.T, opt installOptions) {
+	_, err := os.Stat(binaryPath)
+	require.ErrorIs(t, err, os.ErrNotExist, "binary is already created")
+}
 
-	cmd := exec.Command("/usr/local/bin/otelcol-sumo", "--version")
-	err = cmd.Start()
+func checkBinaryIsRunning(t *testing.T, opt installOptions) {
+	cmd := exec.Command(binaryPath, "--version")
+	err := cmd.Start()
 	require.NoError(t, err)
 
 	code, err := exitCode(cmd)
 	require.NoError(t, err)
 	require.Equal(t, 0, code)
 }
+
+func TestInstallScript(t *testing.T) {
+	for _, tt := range []struct {
+		name       string
+		options    installOptions
+		code       int
+		preChecks  []check
+		postChecks []check
+		preActions []check
```

Comment:
this is place for some pre-test actions like pre-creation of systemd configuration or pre-placing binary